### PR TITLE
[Core] Find global nodal element neighbours - fix const error in centos

### PR DESCRIPTION
**Pull request by ddiezrod**
This was throwing an error in our centos 6 machine. I added a const version of get for ResultsProxy.

image

@KratosMultiphysics/altair

### CODE REVIEW

File: kratos/utilities/pointer_communicator.h [hunk 1]
```diff
@@ -89,6 +89,14 @@ class ResultsProxy
             return mNonLocalData[gp];
     }
 
+    TSendType Get(GlobalPointer<TPointerDataType> const& gp)
```
Comment (philbucher) [inline kratos/utilities/pointer_communicator.h:92]: shouldn't this be:

```suggestion
    const TSendType Get(GlobalPointer<TPointerDataType> const& gp)
```

?

Comment (ddiezrod) [reply kratos/utilities/pointer_communicator.h:92]: > shouldn't this be:
> 
> ?

This is the result from the functor, I think there is no reason why it should be constant. What do you think @RiccardoRossi @pooyan-dadvand ?

Comment (philbucher) [reply kratos/utilities/pointer_communicator.h:92]: I thought it is common to have a const and a non-const version of a fuction, e.g. [here](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/node.h#L705-L712) (I now realized that I might be talking abt sth else, but then I am not sure if the non-const version is needed)

Lets wait for the reply of @RiccardoRossi and @pooyan-dadvand, I am not sure on this one

Comment (RiccardoRossi) [reply kratos/utilities/pointer_communicator.h:92]: @philbucher i don t see a reason for which the returned obj should be const. after retrieving you nay modify it...

regarding the other errror i don t understand hat is happening...

**Review by RiccardoRossi — COMMENTED** [review]
i feliberately avoided the const...

**Comment by ddiezrod** [comment]
@RiccardoRossi you mean the one @philbucher mentions or the ones I added?

**Comment by ddiezrod** [comment]
@RiccardoRossi I think that the problem is that std::sort expects two const objects to compare them according to your defined functor. I guess in newer versions of gcc this is solved to allow you to use non-const versions...

**Comment by RiccardoRossi** [comment]
but then you just need the lamda to have const pointers as input

**Comment by ddiezrod** [comment]
yes, but then you get an error when in lambda function you do `get(gp1)` as there is no version of this get with const input.

**Comment by RiccardoRossi** [comment]
the problem here s that of you have the const in the functor you cannot use.kt to change stuff remotely which was a useful feature.

what compiler version do.you have?

one option would be to pass a copy of the gps in the lambda

**Comment by ddiezrod** [comment]
I think it is gcc 4.9.3 (the default version that centos 6 brings, quite an old one, but the official one supported by altair right now).

I am not sure if I undestand your point, I added an additional "get" in the ResultsProxy class that uses a const reference as input, but you still have the previous "get" that uses a non-const version and therefore the functor can modify the input (just cant be used in this case)

**Review by RiccardoRossi — APPROVED** [review]
ok

**Comment by ddiezrod** [comment]
Thanks!

File: kratos/utilities/pointer_communicator.h (hunk 1 above)
Comment (pooyan-dadvand) [reply kratos/utilities/pointer_communicator.h:92]: @RiccardoRossi the return value should not be const. (You are confused by @philbucher suggestion above) The problem is that you only receive non const GP but what happens if GP is constant and Functor accept it. In this case we need the method added by @ddiezrod 

Answering to @philbucher, The TSendType is the result of the function (for example a copy of pressure) which can be or not constant. So the constantness is reflected in TSendType by the Functor

**Pull request merged**